tools/serverpod_cli/lib/src/generator/open_api/open_api_definition.dart

```diff
@@ -0,0 +1,271 @@
+      );
+      paths.add(pathsObject);
+    }
+  }
+  return paths;
+}
+
+/// Get a set of [OpenAPITag] from protocol definition.
+Set<OpenAPITag> _getTagsFromProtocolDefinition(
+    ProtocolDefinition protocolDefinition) {
+  Set<OpenAPITag> tags = {};
+
+  for (var endpoint in protocolDefinition.endpoints) {
+    var tag = OpenAPITag(
+      name: endpoint.name.camelCase,
+      description: endpoint.documentationComment?.cleanedDartComment,
+    );
+    tags.add(tag);
+  }
+  return tags;
+}
+
+/// Get a set of [OpenAPIComponentSchema] from entities.
+/// example```
+///   Set<SchemaObject> schemas =
+///  _getSchemaObjectFromClassDefinitions(protocolDefinition.entities);
+/// ```
+Set<OpenAPIComponentSchema> _getComponentSchemasFromClassDefinitions(
+    List<SerializableEntityDefinition> entitiesDefinition,
+    List<SerializableEntityDefinition> returnTypeList) {
+  Set<OpenAPIComponentSchema> schemas = {};
+  var entitiesFromMethodReturn = returnTypeList;
+  for (var entityInfo in entitiesDefinition) {
+    // Removes entity that are already present in entitiesDefinition.
+    entitiesFromMethodReturn
+        .removeWhere((e) => e.className == entityInfo.className);
+    if (!entityInfo.serverOnly) {
+      schemas.add(
+        OpenAPIComponentSchema(entityInfo),
+      );
+    }
+  }
+
+  for (var entityInfo in entitiesFromMethodReturn) {
+    if (!entityInfo.serverOnly) {
+      schemas.add(
+        OpenAPIComponentSchema(entityInfo),
+      );
+    }
+  }
+
+  return schemas;
+}
+
+/// Collects entities from [List<EndpointDefinition>] when the type is
+/// 'serializableObjects' to enable validation, ensuring all
+/// serializableObjects are correctly referenced in 'components.schemas'.
+List<SerializableEntityDefinition> _getEntitiesFromEndpointsReturnType(
+  ProtocolDefinition protocolDefinition,
+) {
+  List<SerializableEntityDefinition> returnTypeList = [];
+  for (var endpoint in protocolDefinition.endpoints) {
+    for (var method in endpoint.methods) {
+      var returnType = method.returnType.className == 'Future'
+          ? method.returnType.generics.first
+          : method.returnType;
+      if (returnType.toOpenAPISchemaType !=
+              OpenAPISchemaType.serializableObjects ||
+          returnType.className == 'void' ||
+          returnType.className == 'dynamic') {
+        continue;
+      }
+      SerializableEntityDefinition entity;
+
+      if (returnType.isEnum) {
+        entity = EnumDefinition(
+            fileName: 'undefined',
+            sourceFileName: 'undefined',
+            className: returnType.className,
+            values: [],
+            serverOnly: false);
+      } else {
+        entity = ClassDefinition(
+            fileName: 'undefined',
+            sourceFileName: 'undefined',
+            className: returnType.className,
+            fields: [],
+            serverOnly: false,
+            isException: false);
+      }
+
+      returnTypeList.add(entity);
+    }
```

Comment:
Let's say I have an endpoint returning an `AuthenticationResponse` object. But, in the protocol definition, there is no `AuthenticationResponse` class which is from the auth module. To prevent reference not found errors when generating OpenAPI contents, I've created a placeholder schema with empty data.

Reply:
It seems like a better approach would be to ignore that method call and add a warning when generating the API files.

Reply:
If we support merging from other configurations, such as from the auth module, it will automatically override the placeholder schema.